libraries/package-deps-hash/src/getPackageDeps.ts

```diff
@@ -1,86 +1,148 @@
 import * as child_process from 'child_process';
 import { IPackageDeps } from './IPackageDeps';
 
-const PROCESS_OUTPUT_DELIMITER: string = '///~X~X~X~X~X~X~///';
+export type GitStatusChangeType = 'D' | 'M' | 'A';
 
-/** @public */
+/**
+ * Parses the output of the "git ls-tree" command
+ */
+export function parseGitLsTree(output: string): Map<string, string> {
+  const changes: Map<string, string> = new Map<string, string>();
+
+  if (output) {
+    // Note: The output of git ls-tree uses \n newlines regardless of OS.
+    output.split('\n').forEach(line => {
+
+      // A line is expected to look like:
+      // 100644 blob 3451bccdc831cb43d7a70ed8e628dcf9c7f888c8    src/typings/tsd.d.ts
+
+      if (line) {
+        // Take everything after the "100644 blob", which is just the hash and filename
+        const [hash, filename]: string[] = line.substr(line.indexOf('blob ') + 5).split('\t');
+        changes.set(filename, hash);
+      }
+    });
+  }
+
+  return changes;
+}
+
+/**
+ * Parses the output of the "git status" command
+ */
+export function parseGitStatus(output: string, packagePath: string): Map<string, GitStatusChangeType> {
+  const changes: Map<string, GitStatusChangeType> = new Map<string, GitStatusChangeType>();
+  const filesToHash: string[] = [];
+
+  /*
+  * Typically, output will look something like:
+  * M temp_modules/rush-package-deps-hash/package.json
+  * D package-deps-hash/src/index.ts
+  */
+
+  // If there was an issue with `git ls-tree`, or there are no current changes, processOutputBlocks[1]
+  // will be empty or undefined
+  if (!output) {
+    return changes;
+  }
+
+  // Note: The output of git hash-object uses \n newlines regardless of OS.
+  output
+    .trim()
+    .split('\n')
+    .forEach(line => {
+      const [changeType, filename]: string[] = line.trim().split(' ');
+      /*
+      * changeType == 'D' or 'M' or 'A'
+      * filename == path to the file
+      */
+      if (changeType && filename) {
+        changes.set(filename, changeType as GitStatusChangeType);
+      }
+    });
+
+  return changes;
+}
+
+/**
+ * Takes a list of files and returns the current git hashes for them
+ */
+export function gitHashFiles(filesToHash: string[], packagePath: string): Map<string, string> {
+  const changes: Map<string, string> = new Map<string, string>();
+  if (filesToHash.length) {
+    const hashStdout: string = child_process.execSync(
+      'git hash-object ' + filesToHash.join(' '),
+      { cwd: packagePath }).toString();
+
+    // The result of hashStdout will be a list of file hashes delimited by newlines
+    const hashes: string[] = hashStdout.split('\n');
+
+    filesToHash.forEach((filename, i) => changes.set(filename, hashes[i]));
+  }
+  return changes;
+}
+
+/**
+ * Executes "git ls-tree" in a folder
+ */
+export function gitLsTree(path: string): string {
+  return child_process.execSync(
+    `git ls-tree HEAD -r`,
+    { cwd: path }).toString();
+}
```

Comment:
It's weird that we are executing bare shell commands without knowing where the PATH points.  Maybe out of scope for this PR?